### PR TITLE
Refactors Advanced Rings API/Examples

### PR DESCRIPTION
**Pull request by koolzz**
Currently the advanced ring support in onvm is not clean, this pr attempts to resolve this.

## Summary:
This PR resolves such issues as:

 - Removing unused APIs
 - Removing support for APIs written for advanced rings (see advanced rings scaling) that would never actually be used by the developers, because if developers are using advanced rings the assumption is they will also want to manage their own threads etc.
 - Remove Advanced Rings mode from the Speed tester NF
 - Cleanup and redo adv ring mode in the Scaling NF, the mode now does its own pthread creation instead of relying on onvm scaling APIs. Also make a clear separation between default and advanced ring mode.
 - Because of the adv rings changes some internal nflib APIs were exposed to the NF (`onvm_nflib_start_nf`,  `onvm_nflib_init_nf_init_cfg`, `onvm_nflib_inherit_parent_init_cfg`)

**Usage:**

| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  
| Breaking API changes     |
| Internal API changes     | 👍 
| Usability improvements   |  
| Bug fixes                |
| New functionality        | 👍 
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 

## Merging notes:
 - Dependencies: merge after #117 

**TODO before merging :**
 - [ ] PR is ready for review

## Test Plan:
Still working on it but primarily we will need to test the scaling NF

## Review: 
(optional) << @-mention people who should review these changes >>

(optional) **Subscribers:** << @-mention people who probably care about these changes >>

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@onvm go for it

**Comment by onvm** [comment]
> @onvm go for it

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@onvm Nice how about this one

**Comment by onvm** [comment]
> @onvm Nice how about this one

## CI Message

Your results will arrive shortly

File: examples/scaling_example/scaling.c [hunk 1]
```diff
@@ -230,27 +233,90 @@ packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta, __attribute__((
...
+        for (i = 0; i < num_children; i++) {
+                struct onvm_nf_init_cfg *child_cfg;
+                child_cfg = onvm_nflib_init_nf_init_cfg(nf->tag);
+                /* Prepare state data for the child */
+                child_cfg->service_id = nf->service_id;
+                struct child_spawn_info *child_data = malloc(sizeof(struct child_spawn_info));
+                child_data->child_cfg = child_cfg;
+                child_data->parent = nf;
+                /* Increment the children count because we want the ovnm_nflib to handle children termination */
+                rte_atomic16_inc(&nf->thread_info.children_cnt);
```
Comment (koolzz) [inline examples/scaling_example/scaling.c]: @nks5295 what are we thinking for advanced rings thread managing here? 

We could either use the same concept as we use internally, or we could just declare a global atomic termination flag and just nuke everything when signal is received. 

I'm a fan of the atomic global as we get to showcase how the NF specific signal handler is supposed to be used too.

**Comment by koolzz** [comment]
@onvm linty lint

**Comment by onvm** [comment]
> @onvm linty lint

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@onvm gogogo

**Comment by onvm** [comment]
> @onvm gogogo

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@dennisafa @kevindweb Check this out if you have time, if shows a clean separation between adv and normal rings(at least it should so comment if this needs any changes).

@nks5295 This is good for review now. Updated the description

**Comment by koolzz** [comment]
@onvm comrade _проверь код_

**Comment by onvm** [comment]
> @onvm comrade _проверь код_

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@onvm go for it

**Comment by onvm** [comment]
> @onvm go for it

## CI Message

Your results will arrive shortly

**Comment by koolzz** [comment]
@nks5295 ready for your review 📝

**Pull request merged**